hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsSelector.java

```diff
@@ -0,0 +1,285 @@
+   * functionality, due to https://github.com/apache/hudi/blob/master/style/checkstyle.xml#L270
+   */
+  protected List<List<Message>> createListPartitions(List<Message> singleList, int eachBatchSize) {
+    List<List<Message>> listPartitions = new ArrayList<>();
+
+    if (singleList.size() == 0 || eachBatchSize < 1) {
+      return listPartitions;
+    }
+
+    for (int start = 0; start < singleList.size(); start += eachBatchSize) {
+      int end = Math.min(start + eachBatchSize, singleList.size());
+
+      if (start > end) {
+        throw new IndexOutOfBoundsException(
+            "Index " + start + " is out of the list range <0," + (singleList.size() - 1) + ">");
+      }
+      listPartitions.add(new ArrayList<>(singleList.subList(start, end)));
+    }
+    return listPartitions;
+  }
+
+  /**
+   * delete batch of messages from queue.
+   */
+  protected void deleteBatchOfMessages(
+      AmazonSQS sqs, String queueUrl, List<Message> messagesToBeDeleted) {
+    DeleteMessageBatchRequest deleteBatchReq =
+        new DeleteMessageBatchRequest().withQueueUrl(queueUrl);
+    List<DeleteMessageBatchRequestEntry> deleteEntries = deleteBatchReq.getEntries();
+
+    for (Message message : messagesToBeDeleted) {
+      deleteEntries.add(
+          new DeleteMessageBatchRequestEntry()
+              .withId(message.getMessageId())
+              .withReceiptHandle(message.getReceiptHandle()));
+    }
+    DeleteMessageBatchResult deleteResult = sqs.deleteMessageBatch(deleteBatchReq);
+    List<String> deleteFailures =
+        deleteResult.getFailed().stream()
+            .map(BatchResultErrorEntry::getId)
+            .collect(Collectors.toList());
+    System.out.println("Delete is" + deleteFailures.isEmpty() + "or ignoring it.");
+    if (!deleteFailures.isEmpty()) {
+      log.warn(
+          "Failed to delete "
+              + deleteFailures.size()
+              + " messages out of "
+              + deleteEntries.size()
+              + " from queue.");
+    } else {
+      log.info("Successfully deleted " + deleteEntries.size() + " messages from queue.");
+    }
+  }
+
+  /**
+   * Delete Queue Messages after hudi commit. This method will be invoked by source.onCommit.
+   */
+  public void onCommitDeleteProcessedMessages(
+      AmazonSQS sqs, String queueUrl, List<Message> processedMessages) {
+
+    if (!processedMessages.isEmpty()) {
+
+      // create batch for deletion, SES DeleteMessageBatchRequest only accept max 10 entries
+      List<List<Message>> deleteBatches = createListPartitions(processedMessages, 10);
+      for (List<Message> deleteBatch : deleteBatches) {
+        deleteBatchOfMessages(sqs, queueUrl, deleteBatch);
+      }
+    }
+  }
+
+  /**
+   * Configs supported.
+   */
+  public static class Config {
+    /**
+     * {@value #QUEUE_URL_PROP} is the queue url for cloud object events.
+     */
+    public static final String QUEUE_URL_PROP = "hoodie.deltastreamer.source.queue.url";
```

Comment:
we might need to fix the naming convention of all these configs. 
"hoodie.deltastreamer.cloud.source...." or something on similar lines. Wdyt?

Reply:
Sounds good. I'll change it.